idx/memory/memory.go

```diff
@@ -242,17 +237,17 @@ func (m *MemoryIdx) add(def *schema.MetricDefinition) error {
 		log.Debug("memory-idx: creating branch %s with child %s", branch, nodes[i])
```

Comment:
hmm... ok.
Basically my idea was this, but I must be missing where else it's used...
```
        // Add missing branch nodes
        for ; startPos < len(nodes); startPos++ {
                branch := strings.Join(nodes[0:startPos], ".")
                log.Debug("memory-idx: creating branch %s with child %s", branch, nodes[startPos])
                tree.Items[branch] = &Node{
                        Path:     branch,
                        Children: []string{nodes[startPos]},
                        Defs:     make([]string, 0),
                }
        }

        // Add leaf node
        log.Debug("memory-idx: creating leaf %s", path)
        tree.Items[path] = &Node{
                Path:     path,
                Children: []string{},
                Defs:     []string{def.Id},
        }
        m.DefById[def.Id] = def
        statAddOk.Inc()
        return nil
}
```